kube_apiserver_metrics/tests/test_sli_metrics.py

```diff
@@ -0,0 +1,183 @@
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=4,
+        tags=["sli_name:autoregister-completion", "status:success"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=5,
+        tags=["sli_name:etcd", "status:success"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=5,
+        tags=["sli_name:log", "status:success"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=5,
+        tags=["sli_name:ping", "status:success"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=5,
+        tags=[
+            "sli_name:poststarthook/aggregator-reload-proxy-client-cert",
+            "status:success",
+        ],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        value=5,
+        tags=[
+            "sli_name:poststarthook/apiservice-discovery-controller",
+            "status:success",
+        ],
+    )
+
+    aggregator.assert_all_metrics_covered()
+
+
+def test_check_metrics_slis_transform(aggregator, mock_metrics, mock_request, instance):
+    mock_request.get("http://localhost:6443/metrics/slis", status_code=200)
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    c.check(instance)
+
+    def assert_metric(name, **kwargs):
+        # Wrapper to keep assertions < 120 chars
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, name), **kwargs)
+
+    # Check that no metrics with `name` tag come through
+    assert_metric(
+        "slis.kubernetes_healthcheck",
+        count=0,
+        metric_type=aggregator.GAUGE,
+        tags=["name:autoregister-completion"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        metric_type=aggregator.MONOTONIC_COUNT,
+        count=0,
+        tags=["name:autoregister-completion", "status:error"],
+    )
+
+
+def test_check_metrics_slis_filter_by_type(aggregator, mock_metrics, mock_request, instance):
+    mock_request.get("http://localhost:6443/metrics/slis", status_code=200)
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    c.check(instance)
+
+    def assert_metric(name, **kwargs):
+        # Wrapper to keep assertions < 120 chars
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, name), **kwargs)
+
+    # Check that metrics with type other than `healthz` are filtered out
+    assert_metric(
+        "slis.kubernetes_healthcheck",
+        count=0,
+        metric_type=aggregator.GAUGE,
+        tags=["sli_name:autoregister-completion", "type:readyz"],
+    )
+
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        metric_type=aggregator.MONOTONIC_COUNT,
+        count=0,
+        tags=["sli_name:autoregister-completion", "status:error", "type:readyz"],
+    )
+
+
+@pytest.fixture()
+def mock_request():
+    with requests_mock.Mocker() as m:
```

Comment:
I'd suggest you use the [the fixture](https://github.com/DataDog/integrations-core/blob/e5bc82319897bd769db38cbd36dc622ff95cdb0d/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py#L290) we already have instead of adding a depedency.

Reply:
thanks for pointing this out!